DEV/org.openl.rules.util/src/org/openl/rules/util/Statistics.java

```diff
@@ -68,10 +555,75 @@ public void process(V value) {
             }
         }
 
+        @Override
+        public void process(V y, V x) {
+            if (x != null && y != null) {
+                processNonNull(y, x);
+                counter++;
+            }
+        }
+
         @Override
         public R result() {
             return result;
         }
     }
 
+    private abstract static class InputStats<T extends Number> {
+        T[] x;
+        T[] y;
+
+        public abstract T getAvgX();
+
+        public abstract T getAvgY();
+    }
+
+    private static class InputStatsDouble extends InputStats<Double> {
+        Double avgX;
+        Double avgY;
```

Comment:
What is the reason to use InputStats class? Can be it replaced just by simple arguments?

Reply:
Using this InputStats class we are validating the input for example In case of [1, null, 2] and [1, 2, 3]. These 2 arrays are of equal length but one of the element is null so that pair of input is invalid. After validation the input is further processed as [1,2] and [1, 3]. 
All checks are performed before we start any calculation.